### PR TITLE
Remove prop types

### PR DESCRIPTION
**Pull request by ahuth**
### Summary:

This PR:
- Removes prop-types (noticed it was there while working on #1438)
- Converts the files that had prop-types from JS to TS
- Removes a couple that aren't actually used

### Test Plan:

- Type checking
- Local Storybook

### CODE REVIEW

File: .storybook/components/DesignTokens/Tier1/FontSizes.jsx [hunk 1]
```diff
@@ -1,30 +0,0 @@
-import PropTypes from 'prop-types';
```
Comment (booc0mtaco) [inline .storybook/components/DesignTokens/Tier1/FontSizes.jsx:1]: Were these ones being used b/c they aren't used anywhere?

**Review by booc0mtaco — APPROVED** [review]
nice catch 🎣

File: .storybook/components/DesignTokens/Tier1/FontSizes.jsx (hunk 1 above)
Comment (ahuth) [reply .storybook/components/DesignTokens/Tier1/FontSizes.jsx:1]: That's right, I can't find any usages of them anywhere.

Looking at c8a3bbc28fb688c94f99b0ff73a64bc4cda0b8d8, these were probably part of Big Medium's template.

**Comment by ahuth** [comment]
Thanks

File: .storybook/components/DesignTokens/Tier1/FontSizes.jsx (hunk 1 above)
Comment (jinlee93) [reply .storybook/components/DesignTokens/Tier1/FontSizes.jsx:1]: I went back in time for a while and don't think either of these were being used

**Pull request merged**